src/Symfony/Component/Validator/Constraints/HexaColorValidator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+class HexaColorValidator extends ConstraintValidator
+{
+    private const PATTERN_HTML5 = '/^#[0-9a-f]{6}$/i';
+    private const PATTERN_SHORT = '/^#[0-9a-f]{3}$/i';
+    private const PATTERN_BOTH = '/^#(?:[0-9a-f]{3}){1,2}$/i';
+
+    private const HEXA_COLOR_PATTERNS = [
+        HexaColor::VALIDATION_MODE_HTML5 => self::PATTERN_HTML5,
+        HexaColor::VALIDATION_MODE_SHORT => self::PATTERN_SHORT,
+        HexaColor::VALIDATION_MODE_BOTH => self::PATTERN_BOTH,
+    ];
+
+    private $defaultMode;
+
+    public function __construct(string $defaultMode = HexaColor::VALIDATION_MODE_BOTH)
+    {
+        if (!\in_array($defaultMode, HexaColor::$validationModes, true)) {
```

Comment:
use `if (!isset(self::HEXA_COLOR_PATTERNS[$defaultMode]))` and then you can remove the internal public property